Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector_test.py

```diff
@@ -1,123 +1,152 @@
 import json
-import io
+from urllib.parse import urljoin
+import pytest
+from pytest_mock.plugin import MockerFixture
+from requests_mock.mocker import Mocker as RequestsMock
+from DigitalGuardianARCEventCollector import Client
+
+
+CLIENT_KWARGS = {
+    'verify': False,
+    'proxy': False,
+    'auth_url': 'https://example.com',
+    'base_url': 'https://example.com',
+    'client_id': '11',
+    'client_secret': '22',
+    'export_profile': '33',
+}
 
 
 def util_load_json(path):
-    with io.open(path, mode='r', encoding='utf-8') as f:
+    with open(path, encoding='utf-8') as f:
         return json.loads(f.read())
 
 
-def test_add_time_key_to_events():
+@pytest.fixture
+def authenticated_client(requests_mock: RequestsMock) -> Client:
+    """Fixture to create a Digital Guardian Client instance."""
+    token_url = urljoin(CLIENT_KWARGS['auth_url'], '/as/token.oauth2')
+    requests_mock.post(token_url, json={'access_token': '123', 'expires_in': 10000})
+
+    return Client(**CLIENT_KWARGS)
+
+
+def test_get_access_token(mocker: MockerFixture):
+    """
+    Given:
+        - A valid access token in the integration context
+    When:
+        - Calling Client._get_access_token
+    Then:
+        - Ensure the token in the integration context is returned and no new token is requested.
     """
-       Given:
-           - list of events
-       When:
-           - Calling add_time_key_to_events
-       Then:
-           - Ensure the _time key is added to the events
-       """
-    from DigitalGuardianARCEventCollector import add_time_to_events
+    integration_context_token = {'token': '123', 'valid_until': 2022222222}
+    mocker.patch('DigitalGuardianARCEventCollector.get_integration_context', return_value=integration_context_token)
+    mocker.patch('time.time', return_value=1022222222)
+    get_new_token_request = mocker.patch.object(Client, '_http_request')
 
-    events = util_load_json('test_data/events.json')
-    add_time_to_events(events)
+    client = Client(**CLIENT_KWARGS)
+    access_token = client._get_access_token()
 
-    assert events[0]['_time'] is None
-    assert events[1]['_time'] == "2023-05-23T11:53:11Z"
+    assert access_token == integration_context_token['token']
+    assert get_new_token_request.called is False
 
 
-def test_get_raw_events_command(mocker):
+def test_create_events_for_push():
+    """
+    Given:
+        - list of events and a limit value
+    When:
+        - Calling create_events_for_push
+    Then:
+        - Ensure the _time key is added to the events
+    """
+    from DigitalGuardianARCEventCollector import create_events_for_push
+
+    limit = 2
+    raw_response = util_load_json('test_data/mock_response.json')
+
+    outputted_events = list(create_events_for_push(raw_response, limit))
+
+    assert outputted_events[0]['_time'] is None
+    assert outputted_events[1]['_time'] == "2023-05-23T11:53:11Z"
+    assert len(outputted_events) == limit
```

Comment:
let's use parametrization here, to see that we can get 1,2 events from the list

Reply:
Done in commit e96135ff0130f088a3da78d6d92e024a34457a3d